From 76b0687a547f31b58f4826dd41ec58030b155391 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Tue, 7 Oct 2025 14:41:20 -0500
Subject: [PATCH 325/423] HACK: ntdll: Add sched quantum for ntsync.

It's hard to test this since ubisoft connect doesn't have updates that often... unfortunate
---
 dlls/ntdll/unix/sync.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/ntdll/unix/sync.c b/dlls/ntdll/unix/sync.c
index 5e1b6de7009..9ee0a14e062 100644
--- a/dlls/ntdll/unix/sync.c
+++ b/dlls/ntdll/unix/sync.c
@@ -958,6 +958,12 @@ static NTSTATUS linux_wait_objs( int device, const DWORD count, const int *objs,
             return ret;
         }
 
+        /*
+         * HACK: simulate sched quantum
+         * Ubisoft Connect updater doesn't work without this
+         */
+        if (fsync_simulate_sched_quantum) usleep(0);
+
         return wait_any ? args.index : 0;
     }
     else if (errno == EOWNERDEAD)
-- 
2.52.0

